### PR TITLE
Wait for project restore process to finish before starting with indexing

### LINKED ISSUE #18: Fix flaky CI on Ubuntu
Tests on Ubuntu are regularly failing with this diff
```diff
using DiffPlex.DiffBuilder;
[29](https://github.com/sourcegraph/scip-dotnet/actions/runs/4103369621/jobs/7077473167#step:4:30)
- //      ^^^^^^^^ reference scip-dotnet nuget DiffPlex 1.7.1.0 DiffPlex/
[30](https://github.com/sourcegraph/scip-dotnet/actions/runs/4103369621/jobs/7077473167#step:4:31)
- //               ^^^^^^^^^^^ reference scip-dotnet nuget DiffPlex 1.7.1.0 DiffBuilder/
[31](https://github.com/sourcegraph/scip-dotnet/actions/runs/4103369621/jobs/7077473167#step:4:32)
    using DiffPlex.DiffBuilder.Model;
[32](https://github.com/sourcegraph/scip-dotnet/actions/runs/4103369621/jobs/7077473167#step:4:33)
- //      ^^^^^^^^ reference scip-dotnet nuget DiffPlex 1.7.1.0 DiffPlex/
[33](https://github.com/sourcegraph/scip-dotnet/actions/runs/4103369621/jobs/7077473167#step:4:34)
- //               ^^^^^^^^^^^ reference scip-dotnet nuget DiffPlex 1.7.1.0 DiffBuilder/
[34](https://github.com/sourcegraph/scip-dotnet/actions/runs/4103369621/jobs/7077473167#step:4:35)
- //                           ^^^^^ reference scip-dotnet nuget DiffPlex 1.7.1.0 Model/
[35](https://github.com/sourcegraph/scip-dotnet/actions/runs/4103369621/jobs/7077473167#step:4:36)
        DiffPaneModel Diff()
[36](https://github.com/sourcegraph/scip-dotnet/actions/runs/4103369621/jobs/7077473167#step:4:37)
- //    ^^^^^^^^^^^^^ reference scip-dotnet nuget DiffPlex 1.7.1.0 Model/DiffPaneModel#
[37](https://github.com/sourcegraph/scip-dotnet/actions/runs/4103369621/jobs/7077473167#step:4:38)
            return InlineDiffBuilder.Diff("a", "b");
[38](https://github.com/sourcegraph/scip-dotnet/actions/runs/4103369621/jobs/7077473167#step:4:39)
- //               ^^^^^^^^^^^^^^^^^ reference scip-dotnet nuget DiffPlex 1.7.1.0 DiffBuilder/InlineDiffBuilder#
[39](https://github.com/sourcegraph/scip-dotnet/actions/runs/4103369621/jobs/7077473167#step:4:40)
- //                                 ^^^^ reference scip-dotnet nuget DiffPlex 1.7.1.0 DiffBuilder/InlineDiffBuilder#Diff().
[40](https://github.com/sourcegraph/scip-dotnet/actions/runs/4103369621/jobs/7077473167#step:4:41)
```
This error happens because `dotnet restore` has not been picked up before indexing the test project.

### PR DESCRIPTION
**Pull request by sagi1623**
Hopefully
Fixes #18 
I think this could also be a problem when running the tool in "production"

### CODE REVIEW

**Comment by sagi1623** [comment]
@olafurpg,

Could you please take a look at this PR, it should fix the flaky CI behaviour and in my honest opinion it should be there also for production scenario as it could also happen when someone indexes a project.

File: ScipDotnet/ScipIndexer.cs [hunk 1]
```diff
@@ -34,6 +36,7 @@ private static void Restore(IndexCommandOptions options, FileInfo project)
         };
         options.Logger.LogInformation("$ dotnet {Arguments}", arguments);
         process.Start();
+        process.WaitForExit(DotnetRestoreTimeout);
```
Comment (olafurpg) [inline ScipDotnet/ScipIndexer.cs:39]: Wow, great find! LGTM 👍🏻

File: ScipDotnet/ScipIndexer.cs [hunk 2]
```diff
@@ -13,6 +13,8 @@ namespace ScipDotnet;
 /// </summary>
 public class ScipIndexer
 {
+    private const int DotnetRestoreTimeout = 3000;
```
Comment (olafurpg) [inline ScipDotnet/ScipIndexer.cs:16]: On a slow connection, is it normal for resolve to take longer than 3 seconds? I'd be fine with setting this timeout to a very high value like 5 minutes, and even making it configurable. We can do this in a separate PR, it's fantastic to merge this now to get a stable CI.

**Review by olafurpg — APPROVED** [review]

**Comment by olafurpg** [comment]
Love seeing a green CI on the first run 😄

File: ScipDotnet/ScipIndexer.cs (hunk 2 above)
Comment (sagi1623) [reply ScipDotnet/ScipIndexer.cs:16]: I liked your suggestion, so I have made the following PR: https://github.com/sourcegraph/scip-dotnet/pull/31

**Pull request merged**